### PR TITLE
fs: implement AsRaw{Fd,Handle} for tokio::fs::File (#1364)

### PR DESCRIPTION
**Pull request by Xinkai**
## Motivation
Exposing tokio::fs RawFd/RawHandle

## Solution

Moved file.rs to file/mod.rs to get around mod visibility issue.

### CODE REVIEW

File: tokio-fs/src/file/os/unix/as_raw_fd.rs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+use std::os::unix::io::{AsRawFd, RawFd};
+
+use crate::File;
+
+impl AsRawFd for File {
+    fn as_raw_fd(&self) -> RawFd {
+        let std = self.std.clone();
```
Comment (carllerche) [inline tokio-fs/src/file/os/unix/as_raw_fd.rs]: Why is the clone required?

Comment (Xinkai) [reply tokio-fs/src/file/os/unix/as_raw_fd.rs]: You're right. It doesn't need a clone() here. I must be thinking about something else.

**Comment by carllerche** [comment]
Ah woops. I never circled back here. #1827 added this.

However, #1827 does not include tests. Would you be willing to update the PR to apply the tests to master?

Again, sorry for duplicating work.

**Comment by Xinkai** [comment]
It's also my bad to lose track of this issue, as I have been busy moving from one city to another. 

I will continue working on this (implement tests) when I am done with the rest of the moving routine.

**Pull request closed without merging**